src/routes/game/components/PlaybackControls.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <menu id="playback-controls" data-cy="playback-controls">
+    <UsernameToolTip id="player-username-container" :username="gameStore.playerUsername" />
+    <span id="playback-controls-button-wrapper">
+      <!-- Skip backward -->
+      <v-btn
+        :disabled="!gameHistoryStore.canGoToPreviousState"
+        variant="text"
+        icon="mdi-skip-backward"
+        data-cy="skip-backward"
+        @click="goToState(0)"
+      />
+
+      <!-- Step backward -->
+      <v-btn
+        :disabled="!gameHistoryStore.canGoToPreviousState"
+        variant="text"
+        icon="mdi-step-backward"
+        data-cy="step-backward"
+        @click="goToState(previousGameStateIndex)"
+      />
+
+      <!-- Step forward -->
+      <v-btn
+        :disabled="!gameHistoryStore.canGoToNextState"
+        variant="text"
+        icon="mdi-step-forward"
+        data-cy="step-forward"
+        @click="goToState(currentGameStateIndex + 1)"
+      />
+
+      <!-- Skip forward -->
+      <v-btn
+        :disabled="!gameHistoryStore.canGoToNextState"
+        variant="text"
+        icon="mdi-skip-forward"
+        data-cy="skip-forward"
+        @click="goToState(-1)"
+      />
+    </span>
+  </menu>
+</template>
+
+<script setup>
+import { computed, watch } from 'vue';
+import { useRouter } from 'vue-router';
+import { useGameHistoryStore } from '@/stores/gameHistory';
+import { useGameStore } from '@/stores/game';
+import UsernameToolTip from '@/routes/game/components/UsernameToolTip.vue';
+
+const router = useRouter();
+const gameHistoryStore = useGameHistoryStore();
+const gameStore = useGameStore();
+
+const currentGameStateIndex = computed(() => gameHistoryStore.currentGameStateIndex);
+
+const previousGameStateIndex = computed(() => currentGameStateIndex.value === -1 ?
+  gameHistoryStore.gameStates.length - 2 :
+  currentGameStateIndex.value - 1
+);
+
+function goToState(gameStateIndex) {
+  const route = router.currentRoute.value;
+  router.push({
+    ...route,
+    query: {
+      ...route.query,
+      gameStateIndex,
+    },
+  });  
+}
+
+watch(() => gameHistoryStore.currentGameStateIndex, async (newVal) => {
+  await gameStore.requestGameState(gameStore.id, newVal);
+  gameStore.myPNum = 0;
```

Comment:
Double check if we need this; I think we do since requestGameState doesn't set your pNum if you aren't in the game

Reply:
But maybe it will if `resetPNumIfNull` correctly sets it for spectators